tests/cmdline/commands/test_process.py

```diff
@@ -26,14 +28,218 @@
 from aiida.orm import CalcJobNode, Group, WorkChainNode, WorkflowNode, WorkFunctionNode
 from tests.utils.processes import WaitProcess
 
+FuncArgs = tuple[t.Any, ...]
 
-def await_condition(condition: t.Callable, timeout: int = 1):
+
+def start_daemon_worker_in_foreground_and_redirect_streams(
+    aiida_profile_name: str, log_dir: Path, prepare_func: t.Callable[[FuncArgs], t.Any], prepare_func_args: FuncArgs
+):
+    """Starts a daemon worker and logs its stdout and and stderr streams to a file in the daemon log directory.
+
+    :param aiida_profile_name: The name of the profile the daemon worker should load.
+    :param log_dir: The directory the log of the worker is put
+    :param prepare_func: Called before the worker is started
+    :param prepare_func_args: The arguments passed to the `prepare_func`
+    """
+    import os
+    import sys
+
+    from aiida.engine.daemon.worker import start_daemon_worker
+
+    prepare_func(*prepare_func_args)
+
+    original_stdout = sys.stdout
+    original_stderr = sys.stderr
+
+    try:
+        pid = os.getpid()
+        sys.stdout = open(log_dir / f'worker-{pid}.out', 'w')
+        sys.stderr = open(log_dir / f'worker-{pid}.err', 'w')
+        start_daemon_worker(False, aiida_profile_name)
+    finally:
+        if sys.stdout != original_stdout:
+            sys.stdout.close()
+            sys.stdout = original_stdout
+        if sys.stderr != original_stderr:
+            sys.stderr.close()
+            sys.stderr = original_stderr
+
+
+# We have to define the mock functions globally as we cannot pass local function to a spawn process
+class MockFunctions:
+    @staticmethod
+    def mock_open(_):
+        raise Exception('Mock open exception')
+
+    @staticmethod
+    async def mock_exponential_backoff_retry(*_, **__):
+        from aiida.common.exceptions import TransportTaskException
+
+        raise TransportTaskException
+
+
+@pytest.fixture(scope='function')
+@pytest.mark.usefixtures('started_daemon_client')
+def fork_worker_context(aiida_profile):
+    """Runs daemon worker on a new process with redirected stdout and stderr streams."""
+    import multiprocessing
+
+    from aiida.engine.daemon.client import get_daemon_client
+
+    client = get_daemon_client(aiida_profile)
+    nb_workers = client.get_number_of_workers()
+    client.decrease_workers(nb_workers)
```

Comment:
Why calling `circus` API through decrease_workers again? Could you add a comment here?

Reply:
It is super cool you get `multiprocessing` workers, I tried to replace circus with `multiprocessing` but not able to make it work. Could you open an issue and link to here so we can go back to think if we can fully replace `circus` with stdlib multiprocessing?